src/GraphQL.NewtonsoftJson/GraphQLRequestListJsonConverter.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using GraphQL.Transport;
+using Newtonsoft.Json;
+
+namespace GraphQL.NewtonsoftJson
+{
+    /// <summary>
+    /// A custom JsonConverter for reading or writing a <see cref="GraphQLRequest"/> object.
+    /// </summary>
+    public class GraphQLRequestListJsonConverter : JsonConverter
+    {
+        /// <inheritdoc/>
+        public override bool CanConvert(Type objectType)
+        {
+            return (
+                objectType == typeof(IEnumerable<GraphQLRequest>) ||
+                objectType == typeof(ICollection<GraphQLRequest>) ||
+                objectType == typeof(IReadOnlyCollection<GraphQLRequest>) ||
+                objectType == typeof(IReadOnlyList<GraphQLRequest>) ||
+                objectType == typeof(IList<GraphQLRequest>) ||
+                objectType == typeof(List<GraphQLRequest>) ||
+                objectType == typeof(GraphQLRequest[]));
+        }
+
+        /// <inheritdoc/>
+        public override bool CanRead => true;
+
+        /// <inheritdoc/>
+        public override bool CanWrite => true;
+
+        /// <inheritdoc/>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var requests = (IEnumerable<GraphQLRequest>)value;
```

Comment:
Can save iterator allocation if use custom cases for List and array. 50/50, whatever you want.

Reply:
> Can save iterator allocation if use custom cases for List and array. 50/50, whatever you want.

This is not a GraphQL client, so it is not very likely this method will ever be called.  It replaces no existing functionality in either this or the server project.  Optimizing away one allocation per list of requests serialized is extremely minor anyway, so I'd leave this as-is to minimize complexity.

For `Newtonsoft.Json`, with `CanWrite` set to `false`, the serializer will automatically enumerate and write the individual members, so writing a list is still supported.  So I'll remove the method entirely in this case.

Note that we do have tests that confirm that serializing a list of `GraphQLRequest` instances does indeed serialize correctly, so it can be assured that functionality is still supported even with `CanWrite` as `false`.

`System.Text.Json` does not support read-only converters, so I will leave that code as-is.

Reply:
Done as indicated above

Reply:
OK